dvc/exceptions.py

```diff
@@ -132,10 +132,13 @@ class NotDvcRepoError(DvcException):
 class CyclicGraphError(DvcException):
     def __init__(self, stages):
         assert isinstance(stages, list)
-        msg = "Pipeline has a cycle involving: {}.".format(
-            ", ".join(s.addressing for s in stages)
+        msg = (
+            "Same item(s) defined as both a dependency and an output "
+            "in stage: {stage}."
+        )
+        super().__init__(
+            msg.format(stage=", ".join(s.addressing for s in stages))
```

Comment:
CyclicGraphError may not happen in a single stage, but it usually involves multiple stages.
We may need to find and raise a different error where there are cycles/overlapped output within a single stage.

Reply:
You'd have to look in https://github.com/iterative/dvc/blob/504485ed6e63343d9d201af58a4726420696b6ac/dvc/repo/graph.py#L56 to figure this out.

Reply:
@skshetry  could you explain what is the issue here? 
because if a cycle consists of multiple stages, they will be listed there:

```
set -exu
pushd $TMPDIR

wsp=test_wspace
rep=test_repo

rm -rf $wsp && mkdir $wsp && pushd $wsp
main=$(pwd)
echo $(pwd)
mkdir $rep && pushd $rep

git init

dvc init

echo data >> data

echo "stages:
   #train:
   #  cmd: cp data out
   #  deps:
   #    - data
   #  outs:
   #    - data
   train2:
     cmd: cp data out
     deps:
       - data
     outs:
       - out
   train3:
     cmd: cp out data
     deps:
       - out
     outs:
       - data
" > dvc.yaml

dvc repro -v
```

output: `dvc.exceptions.CyclicGraphError: Same item(s) defined as both a dependency and an output in stage: train3, train2.` 

and for one stage (`train` needs to be uncommented and `train2-3` commented)
output: `dvc.exceptions.CyclicGraphError: Same item(s) defined as both a dependency and an output in stage: train`

Reply:
probably the output might be changed to `.... in stage(s): <list of stages>`